From 9917ab78223c374cb63309d43518c616948b4a80 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 3 Nov 2023 14:31:31 +0100
Subject: [PATCH 0096/1695] plat-stm32mp1: exposes a fastcall SMC watchdog
 service

Default defines watchdog management SMC based service for non-secure
world to manage the watchdog supervised by OP-TEE secure world.
Non-secure world system can leverage this service for example by
enabling a "arm,smc-wdt" compatible node with arm,smc-id=<0xbc000000>
property in its DT.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 86a227cc0f91c7fab43fed317a9faa4d0d00955e plat-stm32mp1: exposes a fastcall SMC watchdog service]
Change-Id: Idaed3d23bd60fb7651b30fc2e379c57cba7ae3f1
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from e1bfa2fd6db43854eb259c0b8e8e2f1977f9b141]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index dada6e6d1..3ae03bb7d 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -232,6 +232,8 @@ CFG_DRIVERS_RSTCTRL ?= $(CFG_STM32_RSTCTRL)
 $(eval $(call cfg-depends-all,CFG_STM32_RSTCTRL,CFG_DRIVERS_RSTCTRL))
 
 CFG_WDT ?= $(CFG_STM32_IWDG)
+CFG_WDT_SM_HANDLER ?= $(CFG_WDT)
+CFG_WDT_SM_HANDLER_ID ?= 0xbc000000
 
 # Platform specific configuration
 CFG_STM32MP_PANIC_ON_TZC_PERM_VIOLATION ?= y
-- 
2.47.2

